cmd/kube-spawn/setup.go

```diff
@@ -102,65 +111,220 @@ func doSetup(numNodes int, baseImage, kubeSpawnDir string) {
+	copyConfigFiles()
+
 	// NOTE: workaround for making kubelet work with port-forward.
 	// Ideally we should solve the port-forward issue by either
 	// creating general add-ons based on torcx, or creating our own
 	// container image, or at least building socat statically on our own.
-	ksExtraDir := ".kube-spawn/extras"
+	ksExtraDir := path.Join(kubeSpawnDir, "extras")
 	if err := os.MkdirAll(ksExtraDir, os.FileMode(0755)); err != nil {
-		log.Fatalf("Unable to create directory %q: %v.", ksExtraDir, err)
+		stderr.Fatalf("Unable to create directory %q: %v.", ksExtraDir, err)
 	}
 	if err := bootstrap.DownloadSocatBin(ksExtraDir); err != nil {
-		log.Fatalf("Error downloading socat files: %s", err)
+		stderr.Fatalf("Error downloading socat files: %s", err)
 	}
 
 	// estimate get pool size based on sum of virtual image sizes.
 	var poolSize int64
 	var err error
 	if poolSize, err = bootstrap.GetPoolSize(baseImage, numNodes); err != nil {
 		// fail hard in case of error, to avoid running unnecessary nodes
-		log.Fatalf("cannot get pool size: %v", err)
+		stderr.Fatalf("cannot get pool size: %v", err)
 	}
 
-	var nodesToRun []string
+	var nodesToRun []nspawntool.Node
 
 	for i := 0; i < numNodes; i++ {
-		name := bootstrap.GetNodeName(i)
-		if !bootstrap.MachineImageExists(name) {
-			if err := bootstrap.NewNode(baseImage, name); err != nil {
-				log.Fatalf("Error cloning base image: %s", err)
+		var node = nspawntool.Node{
+			Name:       bootstrap.GetNodeName(i),
+			K8sVersion: k8srelease,
+			Runtime:    k8sruntime,
+		}
+		if !bootstrap.MachineImageExists(node.Name) {
+			if err := bootstrap.NewNode(baseImage, node.Name); err != nil {
+				stderr.Fatalf("Error cloning base image: %s", err)
 			}
 		}
-		if bootstrap.IsNodeRunning(name) {
+		if bootstrap.IsNodeRunning(node.Name) {
 			continue
 		}
-		nodesToRun = append(nodesToRun, name)
+		nodesToRun = append(nodesToRun, node)
 	}
 	if len(nodesToRun) == 0 {
-		log.Printf("No nodes to create. stop")
+		stdout.Printf("No nodes to create. stop")
 		os.Exit(1)
 	}
 
 	if err := bootstrap.EnlargeStoragePool(poolSize); err != nil {
-		log.Printf("Warning: cannot enlarge storage pool: %v", err)
+		stdout.Printf("Warning: cannot enlarge storage pool: %v", err)
 	}
 
-	for _, name := range nodesToRun {
-		if err := nspawntool.RunNode(k8srelease, name, kubeSpawnDir); err != nil {
-			log.Fatalf("Error running node: %s", err)
+	for _, node := range nodesToRun {
+		if err := node.Run(kubeSpawnDir, rktBinDir, rktletBinDir); err != nil {
+			stderr.Fatalf("Error running node: %v", err)
 		}
 
-		if err := nspawntool.RunBootstrapScript(name); err != nil {
-			log.Fatalf("Error running bootstrap script: %s", err)
+		if err := node.Bootstrap(); err != nil {
+			stderr.Fatalf("Error running bootstrap script: %v", err)
 		}
 
-		log.Printf("Success! %s started.", name)
+		stdout.Printf("Success! %s started.", node.Name)
+	}
+
+	stdout.Print("All nodes are running. Use machinectl to login/stop/etc.")
```

Comment:
Nit: "Use machinectl to login ..." is confusing as long as `machinectl login ...` doesn't work :)

Personally, I would prefer better documentation in the README (for example, explain common task such as up, start, stop, delete, etc).